tests/purs/passing/BlockStringEdgeCases.purs

```diff
@@ -0,0 +1,30 @@
+module Main where
+
+import Prelude
+import Effect.Console (log)
+import Test.Assert (assert')
+
+data Tuple a b = Tuple a b
+derive instance tupleEq :: (Eq a, Eq b) => Eq (Tuple a b)
+
+main = do
+  assert' "empty string" ("""""" == "")
+  assert' "quote" (""""""" == "\"")
+  assert' "starts with quote" (""""x""" == "\"x")
+  assert' "ends with quote" ("""x"""" == "x\"")
+  assert' "two quotes" ("""""""" == "\"\"")
+  assert' "starts with two quotes" ("""""x""" == "\"\"x")
+  assert' "ends with two quotes" ("""x""""" == "x\"\"")
+  assert' "starts and ends with two quotes" ("""""x""""" == "\"\"x\"\"")
+  assert' "mixture 1" ("""""x"y""z"""" == "\"\"x\"y\"\"z\"")
+  assert' "mixture 2" ("""x"y""z""" == "x\"y\"\"z")
+
+  -- These last tests are more about forbidding certain raw string literal
+  -- edge cases than about wanting to support mashing string literals against.
+  -- each other, which is techically legal but generally, if not universally,
+  -- a bad idea.
+  assert' "too many quotes 1" (Tuple """"""""" " == Tuple "\"\"" " ")
```

Comment:
It occurs to me that it might be worth trying to update the PureScript syntax highlighting because the syntax highlighting here is wrong, presumably? This is what it looks like for me:

![Screenshot from 2020-12-16 03-04-06](https://user-images.githubusercontent.com/1270186/102299752-697f9c80-3f4b-11eb-86f9-ac869ee19669.png)



The highlighting used on GitHub is defined in https://github.com/purescript-contrib/atom-language-purescript if I remember correctly. This is not a blocker for merging, just thought it's worth mentioning.

Reply:
Ah yes, I see what's going on here now; the highlighting is definitely wrong. It wouldn't compile if the parser understood the boundaries of these string literals the same way this highlighter did.